kernel/constr.ml

```diff
@@ -1071,12 +1050,8 @@ let constr_ord_int f t1 t2 =
     | CoFix _, _ -> -1 | _, CoFix _ -> 1
     | Proj (p1,_r1,c1), Proj (p2,_r2,c2) -> compare [(Projection.CanOrd.compare, p1, p2); (f, c1, c2)]
     | Proj _, _ -> -1 | _, Proj _ -> 1
-    | Int i1, Int i2 -> Uint63.compare i1 i2
-    | Int _, _ -> -1 | _, Int _ -> 1
-    | Float f1, Float f2 -> Float64.total_compare f1 f2
-    | Array(_u1,t1,def1,ty1), Array(_u2,t2,def2,ty2) ->
-      compare [(Array.compare f, t1, t2); (f, def1, def2); (f, ty1, ty2)]
-    | Array _, _ -> -1 | _, Array _ -> 1
+    | PVal v1, PVal v2 -> CPrimVal.compare f v1 v2
+    (*| PVal _, _ -> -1 | _, PVal _ -> 1*)
```

Comment:
Not sure I understand why this is commented out? how is the matching complete?

Reply:
I was initially surprised as well, that's why I left the comment, but these patterns are not reachable. Basically, for all constructors there are branches `| C _, _ -> -1 | _, C _ -> 1`, so all cases of the form `C _, PVal _` and `PVal _, C _` are indeed taken care of.

The reason why the code looked different before was that `Float` did not `Float _, _` and `_, Float _` cases (probably by mistake?).